### DISCUSSION TITLE
UnsatisfiedResolutionException

### DISCUSSION THREAD

**Discussion by asanglard** [quarkiverse/quarkus-cxf#408] [Q&A]
Hello,

when I try to start my app, this error appears

java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.arc.deployment.ArcProcessor#validate threw an exception: javax.enterprise.inject.spi.DeploymentException: javax.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type com.acme.MyService and qualifiers [@CXFClient(value = "MyService")]

Caused by: javax.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type com.orange.holmes.cfx.com.acme.MyService and qualifiers [@CXFClient(value = "MyService")]
	- java member: MySoapClient#MyService
	- declared on CLASS bean [types=[java.lang.Object, MySoapClient], qualifiers=[@Default, @Any], target=MySoapClient]
	at io.quarkus.arc.processor.Beans.resolveInjectionPoint(Beans.java:428)
	at io.quarkus.arc.processor.BeanInfo.init(BeanInfo.java:524)
	at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:260)
	... 13 more

I have previously generated cfx source code with maven plugin.

Any idea to solve this problem ?

**Comment by shumonsharif** [comment]
Hello @asanglard - are you able to provide some additional details ... these will be helpful to provide suggestions:

- How are you injecting the `@CXFClient` ... can you share that code?
- How is the client defined in your `application.properties` file ... can you share the relevant properties?

**Reply by asanglard** [reply]
thanks for your quick reply.

here is the code (I changed class name for privacy concern):

About CFX client injection:
```
    @Inject @CXFClient("myService")
    SoapTechnicalService soapTechnicalService;
```
About properties:
```
quarkus.cxf.client."myService".service-interface=com.acme.SoapTechnicalService
quarkus.cxf.client."myService".client-endpoint-url=http://localhost:9081
```

**Reply by shumonsharif** [reply]
Hi @asanglard Only immediate issue I see is that in the exception you posted initially, the `@CXFClient` injection appears to be named "MyService" (starting with a capital M).  In the configurations you posted above, you have a lowercase m for "myService".  The name is case sensitive and should match between the `application.properties` and the `@CXFClient` injection point.

Let me know in case you're still having an issue, and we can keep investigating.

**Reply by asanglard** [reply]
Hi @shumonsharif 

the M in capital for MyService is unfortunately a typo when I changed the class and field names (for privacy purpose) before to post the logs. In the code, I used the same case.

The root cause is somewhere else.

I will continue to dig. Don’t hesitate if you have an other idea.

Aurélien

**Reply by famod** [reply]
Hi @asanglard, it would certainly help if you could upload a small reproducer somewhere.

**Reply by asanglard** [reply]
Hi @famod 

Excellent idea. I will do it on Thursday.

Best regards,
Aurelien

**Comment by asanglard** [comment]
hi @famod @shumonsharif 

we finally found the root cause of our problem.

A set of file is generated using cxf-codegen-plugin maven plugin. This set contains various files including:
- a java interface named soapTechnical annotated with @WebService. This is related to : <wsdl:binding name="ServiceSoapBinding" type="impl:soapTechnical"> in our wsdl.
- a java class named soapTechnicalService annotated with @WebServiceClient. This is related to : <wsdl:service name="soapTechnicalService"> in our wsdl.

Our CXFClient was typed as the soapTechnicalService and not the interface soapTechnical. Using the interface, it works properly, as referred in the property name service-**interface**.

The documentation which gives an example suffixed Service (FruitWeb**Service**) misleads us.

Some suggestions :
- I don't know if the error generated in the log may be more explicit in case someone else make the same error. Like a missing WebService annotation on the service-interface specified...
- In the doc, maybe you can also explicit the SEI acronym (Service Endpoint **Interface**)

**Reply by shumonsharif** [reply]
Thanks @asanglard for the feedback, both are excellent points, will look into addressing both in the next release!  I created issue #410 to track this.

**Reply by shumonsharif** [reply]
Hi @asanglard I dug in a little for this issue, and it appears the error you were seeing occurs well before it hits the extension code (ie. it originates from Quarkus itself).  Given this, I'm afraid there's not much we can do for logging when using the wrong configuration or an invalid SEI.

I updated the documentation as suggested though, to spell out Service Endpoint Interface.  Thanks again for the suggestions!
